tss-esapi/src/structures/buffers.rs

```diff
@@ -157,17 +155,49 @@ pub mod digest {
         type Error = Error;
 
         fn try_from(value: Digest) -> Result<Self> {
-            if value.value().len() != 64 {
+            if value.len() != 64 {
                 return Err(Error::local_error(WrapperErrorKind::WrongParamSize));
             }
 
             let mut result = [0; 64];
 
-            result.copy_from_slice(value.value());
+            result.copy_from_slice(value.as_bytes());
 
             Ok(result)
         }
     }
+
+    impl From<[u8; 20]> for Digest {
+        fn from(mut value: [u8; 20]) -> Self {
+            let value_as_vec = value.to_vec();
+            value.zeroize();
+            Digest(value_as_vec.into())
+        }
+    }
```

Comment:
Perhaps these could be changed into a macro + 4 calls to it? I'm sure other buffers could benefit from it as well.